### PR TITLE
Fix bug in checkDataPart

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Fix bug in check function of *MergeTree engines family. Now it doesn't fail in case when we have equal amount of rows in last granule and last mark (non-final).

### CODE REVIEW

File: dbms/src/Storages/MergeTree/checkDataPart.cpp [hunk 1]
```diff
@@ -379,7 +381,8 @@ MergeTreeData::DataPart::Checksums checkDataPart(
             size_t read_size = tmp_column->size();
             column_size += read_size;
 
-            if (read_size < rows_after_mark || mark_num == adaptive_index_granularity.getMarksCount())
```
Comment (KochetovNicolai) [inline dbms/src/Storages/MergeTree/checkDataPart.cpp]: Why do we remove condition `read_size < rows_after_mark`? Isn't it necessary for non-adaptive index granularity?

Comment (alesapin) [reply dbms/src/Storages/MergeTree/checkDataPart.cpp]: No, in non-adaptive table we must read as much marks as in `adaptive_index_granularity.getMarksCountWithoutFinal`.

Comment (alesapin) [reply dbms/src/Storages/MergeTree/checkDataPart.cpp]: But I'll add a test for it.

Comment (KochetovNicolai) [reply dbms/src/Storages/MergeTree/checkDataPart.cpp]: Can you please also add test with number of rows  != 8192 * N (for non-adaptive table)?

**Review by KochetovNicolai — APPROVED** [review]

**Comment by alesapin** [comment]
`00094_union_race_conditions_5` not related to changes.

**Pull request merged**